forms/scraper/keyword_result_form_test.go

```diff
@@ -0,0 +1,321 @@
+					Expect(formValidation.Errors[0].Message).To(Equal("All Link list must be valid URL"))
+				})
+			})
+		})
+	})
+
+	Describe("#Create", func() {
+		Context("given valid params", func() {
+			It("returns a keyword record", func() {
+				user := FabricateUser(faker.Email(), faker.Password())
+				keyword := faker.Word()
+				visitUrl := fmt.Sprintf("https://www.google.com/search?q=%s&lr=lang_en", keyword)
+				rawHtml := faker.Paragraph()
+				form := form.KeywordResultForm{
+					Keyword:  keyword,
+					VisitUrl: visitUrl,
+					RawHtml:  rawHtml,
+					User:     user,
+				}
+
+				keywordRecord, _ := form.Save()
+
+				Expect(keywordRecord).NotTo(BeNil())
+				Expect(keywordRecord.Id).To(BeNumerically(">", 0))
+			})
+
+			It("does NOT produce any errors", func() {
+				user := FabricateUser(faker.Email(), faker.Password())
+				keyword := faker.Word()
+				visitUrl := fmt.Sprintf("https://www.google.com/search?q=%s&lr=lang_en", keyword)
+				rawHtml := faker.Paragraph()
+				form := form.KeywordResultForm{
+					Keyword:  keyword,
+					VisitUrl: visitUrl,
+					RawHtml:  rawHtml,
+					User:     user,
+				}
+
+				_, errors := form.Save()
+
+				Expect(len(errors)).To(BeZero())
+			})
+		})
+
+		Context("given INVALID params", func() {
+			Context("given NO keyword", func() {
+				It("does NOT return a keyword record", func() {
+					user := FabricateUser(faker.Email(), faker.Password())
+					visitUrl := "https://www.google.com/search?q=&lr=lang_en"
+					rawHtml := faker.Paragraph()
+					form := form.KeywordResultForm{
+						VisitUrl: visitUrl,
+						RawHtml:  rawHtml,
+						User:     user,
+					}
+
+					keywordRecord, errors := form.Save()
+
+					Expect(keywordRecord).To(BeNil())
+					Expect(errors[0].Error()).To(Equal("Keyword can not be empty"))
+				})
+			})
+
+			Context("given keyword length is over than 128", func() {
+				It("does NOT return a keyword record", func() {
+					user := FabricateUser(faker.Email(), faker.Password())
+					keyword := faker.Word()
+					visitUrl := fmt.Sprintf("https://www.google.com/search?q=%s&lr=lang_en", keyword)
+					rawHtml := faker.Paragraph()
+					form := form.KeywordResultForm{
+						Keyword:  keyword + strings.Repeat("*", 128),
+						VisitUrl: visitUrl,
+						RawHtml:  rawHtml,
+						User:     user,
+					}
+
+					keywordRecord, errors := form.Save()
+
+					Expect(keywordRecord).To(BeNil())
+					Expect(errors[0].Error()).To(Equal("Keyword maximum size is 128"))
+				})
+			})
+
+			Context("given NO visit URL", func() {
+				It("does NOT return a keyword record", func() {
+					user := FabricateUser(faker.Email(), faker.Password())
+					keyword := faker.Word()
+					rawHtml := faker.Paragraph()
+					form := form.KeywordResultForm{
+						Keyword: keyword,
+						RawHtml: rawHtml,
+						User:    user,
+					}
+
+					keywordRecord, errors := form.Save()
+
+					Expect(keywordRecord).To(BeNil())
+					Expect(errors[0].Error()).To(Equal("VisitUrl can not be empty"))
```

Comment:
Same mistake as raised before:

```suggestion
					Expect(errors[0].Error()).To(Equal("VisitUrl cannot be empty"))
```

`cannot` must be written as a single word. Please recheck everywhere.